shpc/main/modules/__init__.py

```diff
@@ -147,7 +147,8 @@ def add(self, sif, module_name, **kwargs):
         module_name = self.add_namespace(module_name)
         template = self._load_template(self.templatefile)
         modulefile = os.path.join(self.settings.module_base, module_name.replace(":", os.sep), self.modulefile)
-        self.container.add(sif, module_name, modulefile, template, **kwargs)
+        wrapper_template = self._load_template('.'.join([self.container.templatefile,'sh']))
```

Comment:
I'm trying to think - given that we have other kinds of wrapper scripts - how would this be extended (e.g. right now self.container.templatefile is hard coded).

Reply:
ok, first very simple pass of abstraction here,
I have defined an attribute to the `ModuleBase` class, called `wrappertemplatefile`, so that now the assignment above reads:
```
wrapper_template = self._load_template(self.wrappertemplatefile)
```
Still specific, but a bit cleaner.
In subsequent passes, selection of the template file may become more articulate